phpBB/phpbb/storage/adapter/local.php

```diff
@@ -195,4 +196,57 @@ protected function ensure_directory_exists($path)
 			$this->create_dir($path);
 		}
 	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function read_stream($path)
+	{
+		$stream = @fopen($this->root_path . $path, 'rb');
+
+		if (!$stream)
+		{
+			throw new exception('STORAGE_CANNOT_OPEN_FILE', $path);
+		}
+
+		return $stream;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function write_stream($path, $resource)
+	{
+		if ($this->exists($path))
+		{
+			throw new exception('STORAGE_FILE_EXISTS', $path);
+		}
+
+		$stream = @fopen($this->root_path . $path, 'w+b');
+
+		if (!$stream)
+		{
+			throw new exception('STORAGE_CANNOT_CREATE_FILE', $path);
+		}
+
+		if (stream_copy_to_stream($resource, $stream) === false)
+		{
+			throw new exception('STORAGE_CANNOT_COPY_RESOURCE');
+		}
+	}
+
+	public function file_properties($path)
+	{
+		return [];
+	}
+
+	public function file_size($path)
+	{
+		return filesize($this->root_path . $path);
+	}
+
+	public function file_mimetype($path)
+	{
+		return mime_content_type($this->root_path . $path);
```

Comment:
I am looking that this function is obsolete, but i dont find much information about Fileinfo